vector/src/main/java/im/vector/app/features/crypto/quads/SharedSecureStorageViewModel.kt

```diff
@@ -125,15 +126,17 @@ class SharedSecureStorageViewModel @AssistedInject constructor(
                     copy(
                             hasPassphrase = true,
                             ready = true,
-                            step = SharedSecureStorageViewState.Step.EnterPassphrase
+                            step = if (initialState.step == SharedSecureStorageViewState.Step.ResetAll) SharedSecureStorageViewState.Step.ResetAll
+                            else SharedSecureStorageViewState.Step.EnterPassphrase
```

Comment:
I find this a bit strange.
Wouldn't it be better to test if reset mode before trying to get the `keyResult`?
If state is not reset try to get keyResult and set the hasPassphrase/ready?